subsystem-test/framework/src/main/java/org/jboss/as/subsystem/test/AbstractKernelServicesImpl.java

```diff
@@ -144,6 +145,22 @@ public ModelTestModelControllerService create (Extension mainExtension, Controll
         return kernelServices;
     }
 
+    private static void addDependencyViaReflection(final ServiceBuilder builder, final ServiceName dependencyName) {
```

Comment:
why do you need to add/require the dependency by reflection when the original code didn't?

Reply:
@jmesnil Because there are some transformer tests testing legacy EAP installations using AbstractKernelServicesImpl class and previous EAP versions didn't have MSC with ServiceBuilder.requires() method. It was introduced since EAP 7.1.

Reply:
@jmesnil added